### DISCUSSION TITLE
ONVIF Client Test Tool Profile T Tests - Motion Alarm and Get Digital Inputs tests marked as not passed

### DISCUSSION THREAD

**Discussion by vdelchev** [onvif/testspecs#12] [Q&A] [answered]
[Network trace captures.zip](https://github.com/onvif/testspecs/files/7407877/Network.trace.captures.zip)

Hello,

I am working on the conformance process for ONVIF Profile T with our VMS as an ONVIF client. **What has to be done in order for an ONVIF client to pass tests related to receiving pull point event notifications such as the ones in the Mandatory Motion Alarms test and the Conditional Get Digital Inputs test?**

ONVIF Client Test Tool 21.06 marks the Mandatory Motion Alarm test as not passed despite having provided network trace capture file with pull point event handling and tns1:VideoSource/MotionAlarm notifications.

In the provided attachment I have added two network trace capture files for AXIS P5654-E - one for the manual device discovery with our VMS (10.92.196.233) and AXIS P5654-E (10.92.196.77), and one for the Motion Alarm test. In the second capture can be seen packets related to pull point event handling and the VMS successfully receiving tns1:VideoSource/MotionAlarm notifications.

As seen at "OnvifClientTestTool.png" when I run ONVIF Client Test Tool 21.06 with these two captures I successfully pass the EVENTHANDLING-1PULLPOINT part of the Motion Alarm test but I also see "Passed with Devices: 0". On the right in the Specification tab about MotionAlarmPullPoint can be seen the following message: "blank#tc.MotionAlarm.MotionAlarmPullPoint".

Going over the "ONVIF® Profile T Client Test Specification Version 21.06" I cannot find the reason why the Motion Alarm test fails to pass despite as seen at the provided "MotionAlarm.pcapng" VMS executing pull point event handling and receiving tns1:VideoSource/MotionAlarm notifications.

I am also experiencing similar issues with the Conditional Get Digital Inputs test done with BOSCH AUTODOME inteox 7000i (10.92.196.102) for which I have also provided network trace capture files. I successfully pass the EVENTHANDLING-1 PULLPOINT and GETDIGITALINPUTS-1 GET DIGITAL INPUTS parts of the test but the Get Digital Inputs test is marked as not passed despite having also captured packets with tns1:Device/Trigger/DigitalInput notifications. Just like for MotionAlarmPullPoint, I see a blank message at the Specification tab also for DeviceTriggerDigitalInputNotification: "blank#tc.GetDigitalInputs.DeviceTriggerDigitalInputNotification".

Sincerely,
Venelin Delchev

**Comment by Johan-Svensk** [comment]
Hi,
I have no personal experience with running the Client test tool, but the general ONVIF rules for client conformance states that a client shall test with 3 different device manufacturers products in order to claim conformance (see Fig. 1: ONVIF Conformance Process flow chart in [ONVIF Conformance Process Specification](https://www.onvif.org/wp-content/uploads/2019/10/ONVIF-Conformance-Process-Specification_v5-0.pdf?ccc393&ccc393)). From what I can see from your screenshot (for motion alarm) you have only used one product (hence the reason that you get 'not enough devices'-message from the tool). You can also see that it says 'Required Number of Devices: 3'. 
I hope it helps.

Regards,
Johan

**Reply by vdelchev** [reply]
Hello Johan,

Thank you for your answer. It looks like you have not understood my question. Do you have any suggestion why the Get Digital Inputs test is marked as not passed (You can see that it says 'Required Number of Devices: 1')?

![image](https://user-images.githubusercontent.com/93116669/139671973-147d86b0-7e82-4339-825f-462190380b02.png)

**I have mentioned absolutely nothing about 'not enough devices'.** 'Required Number of Devices: 3' means that you have to use three devices to pass the profile certification. I am not asking about this. **I am asking about the requirements to pass the mandatory Motion Alarm and conditional Get Digital Inputs tests**. 

As seen at the screen capture above you can have a 'Required Number of Devices: 3', use one device to find out if you are able to pass the test with it, and get a 'Passed with devices: 1'.

![image](https://user-images.githubusercontent.com/93116669/139673611-b7c01a84-ea7a-45ff-adf5-cd941e93c97e.png)

As seen at the screen capture above **even with three devices from three different manufacturers the Motion Alarm test is marked as not passed**. I successfully pass the **EVENTHANDLING-1PULLPOINT** part of the Motion Alarm test with the three devices but I also see "**Passed with Devices: 0**". I cannot find any specification about the requirements to pass **MotionAlarmPullPoint** at "[ONVIF® Profile T Client Test Specification Version 21.06](https://www.onvif.org/wp-content/uploads/2021/06/ONVIF_Profile_T_Client_Test_Specification_21.06.pdf?ccc393&ccc393)". There is also no information at the ONVIF Client Test Tool, just blank#tc.MotionAlarm.MotionAlarmPullPoint as seen in the screen capture.

Here is the error I get in the Client Conformance Audit Log of the ONVIF Client Test Tool:
![image](https://user-images.githubusercontent.com/93116669/139676488-284394ab-2b36-4630-b1c5-bf11ac614702.png)
How can a pass the test for 'Retrieve MotionAlarm Notifications'? Where can I find any specification for this?

The same is valid for Get Digital Inputs - where can I find information for passing **DeviceTriggerDigitalInputNotification**? There is no information about it at "[ONVIF® Profile T Client Test Specification Version 21.06](https://www.onvif.org/wp-content/uploads/2021/06/ONVIF_Profile_T_Client_Test_Specification_21.06.pdf?ccc393&ccc393)". There is also no information at the ONVIF Client Test Tool, just blank#tc.GetDigitalInputs.DeviceTriggerDigitalInputNotification.

In the Client Conformance Audit Log of the ONVIF Client Test Tool I get the following warning: 'Warning: Conditional feature DigitalInput Changed Notification has not passed'. How can a pass the test for 'DigitalInput Changed Notification '? Where can I find any specification for this?

Please elaborate on the requirements for passing the **MotionAlarmPullPoint** part of the mandatory MotionAlarm test and the .**DeviceTriggerDigitalInputNotification** part of the conditional Get Digital Inputs test.

Sincerely,
Venelin Delchev

**Reply by Johan-Svensk** [reply]
Sorry if I misunderstood your question. I have asked the Client Test Tool vendor to look into you question and reply back to you.

Regards,
Johan

**Comment by madhurao68** [comment]
Hi,
Could you please provide all the network traces? This will help us to find the reason for the failures. Thanks!

**Comment by Maria-Ver** [comment, marked as answer]
Hello!

For  Motion Alarm one of the pass criteria is "Client is able to retrieve tns1:VideoSource/MotionAlarm notifications if Device supports Motion Alarm feature".
For Get Digital Inputs it is "Client is able to monitor the state of the input pins using tns1:Device/Trigger/DigitalInput event if device supports DigitalInputs feature".

To indicate their support go to the "Configuration" - "Client Settings" - "Supported events" tab and check them in the tree:
![image](https://user-images.githubusercontent.com/93972773/140886812-1d24b052-34e6-433d-9ca1-8225aa2c728c.png)

After that tests should pass. These are the results with your traces:
![image](https://user-images.githubusercontent.com/93972773/140886835-62971795-fb06-4662-996a-ad533e47a1c3.png)

![image](https://user-images.githubusercontent.com/93972773/140886852-bfe7d963-0a1d-4d20-be7c-fbe60cf0d8d0.png)

The functionality of jumping to the test spec when clicking on the Events (and Metadata) nodes is not implemented in the current version, but if you click on the Feature node you will be shown all requirements:
![image](https://user-images.githubusercontent.com/93972773/140886872-a17fafc2-f40a-4807-ae90-fcce919cc655.png)

I hope it helps.

**Reply by vdelchev** [reply]
Hello.

Thank you very much! I followed your advise and now both Motion Alarm and Get Digital Inputs tests are marked as passed. I thought that ONVIF Client Test Tool automatically detected which features are supported based on the network trace captures and provided device features list.

Sincerely,
Venelin Delchev
